gooddata-java/src/test/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigServiceAT.java

```diff
@@ -0,0 +1,126 @@
+    public void createClientSettingViaProjectsApi() {
+        gd.getHierarchicalConfigService()
+                .upsertProjectSetting(project, new SettingItem(FIRST_SETTING_ITEM, "true", "client"));
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "createProjectSetting")
+    public void projectSettings() {
+        final Settings settings = gd.getHierarchicalConfigService().projectSettings(project);
+
+        assertThat(settings, hasItems(
+                new SettingItem(SECOND_SETTING_ITEM, "false", "project"),
+                new SettingItem(FIRST_SETTING_ITEM, "true", "project")));
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "projectSettings")
+    public void getProjectSetting() {
+        final SettingItem settingItem =
+                gd.getHierarchicalConfigService().getProjectSetting(project, SECOND_SETTING_ITEM);
+        checkSettingItem(settingItem, SECOND_SETTING_ITEM, "false");
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "getProjectSetting")
+    public void updateProjectSetting() {
+        final SettingItem settingItem = gd.getHierarchicalConfigService().getProjectSetting(project,
+                FIRST_SETTING_ITEM);
+
+        // update setting value
+        settingItem.setValue("false");
+        final SettingItem disabledSettingItem = gd.getHierarchicalConfigService().upsertProjectSetting(project, settingItem);
+        checkSettingItem(disabledSettingItem, FIRST_SETTING_ITEM, "false");
+
+        // enable again
+        settingItem.setValue("true");
+        final SettingItem enabledSettingItem = gd.getHierarchicalConfigService().upsertProjectSetting(project, settingItem);
+        checkSettingItem(enabledSettingItem, FIRST_SETTING_ITEM , "true");
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "updateProjectSetting")
+    public void changeProjectSetting() {
+        final HierarchicalConfigService hierarchicalConfigService = gd.getHierarchicalConfigService();
+        SettingItem settingItem = hierarchicalConfigService.getProjectSetting(project, FIRST_SETTING_ITEM);
+        String value = String.valueOf(!settingItem.isEnabled());
+        settingItem.setValue(value);
+        hierarchicalConfigService.upsertProjectSetting(project, settingItem);
+        checkSettingItem(settingItem, FIRST_SETTING_ITEM, value);
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "changeProjectSetting")
+    public void deleteProjectSetting() throws IllegalAccessException, InvocationTargetException, InstantiationException,
+            ClassNotFoundException, NoSuchMethodException, NoSuchFieldException {
+        SettingItem updateSetting = new SettingItem(FIRST_SETTING_ITEM, "false", "project");
+        setLinks(updateSetting, "/gdc/projects/" + project.getId() + "/config/" + FIRST_SETTING_ITEM);
+        final SettingItem updated = gd.getHierarchicalConfigService().upsertProjectSetting(project, updateSetting);
```

Comment:
No need to recreate if would depend on `updateProjectSetting`

Reply:
It is needed, cause I need to set the `links` member.

Reply:
Well I would get the existing item from API (or pass from previous method) instead of creating new object ;)